narwhals/expression.py

```diff
@@ -74,6 +74,59 @@ def cast(
         self,
         dtype: Any,
     ) -> Expr:
+        """
+        Redefine an objects data type.
+
+        Arguments:
+            dtype: Data type that the object will be cast into.
+
+        Examples:
+                >>> import pandas as pd
+                >>> import narwhals as nw
+                >>> from datetime import date
+                >>> df_pd = pd.DataFrame(
+                >>> {"foo": [1, 2, 3],
+                >>> "bar": [6.0, 7.0, 8.0],
+                >>> "ham": [date(2020, 1, 2), date(2021, 3, 4), date(2022, 5, 6)],})
+                >>> df_pl = pl.DataFrame
+                >>> {"foo": [1, 2, 3],
+                >>> "bar": [6.0, 7.0, 8.0],
+                >>> "ham": [date(2020, 1, 2), date(2021, 3, 4), date(2022, 5, 6)],})
+
+                Let's make it data frame agnostic:
+
+                >>> df = nw.from_native(df_pd)
+
+                Cast specific frame columns to the specified dtypes:
+
+                >>> df = df.cast({"foo": pl.Float32, "bar": pl.UInt8})
```

Comment:
close 😉  we need to use `Expr.cast` here, not `DataFrame.cast`

something like
```
df = df.select(nw.col('foo').cast(nw.Float32), nw.col('bar').cast(nw.UInt8))
```